src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -200,8 +200,9 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
 
             // If we have something like M(double? x = 1) then the expression we'll get is (double?)1, which
             // does not have a constant value. The constant value we want is (double)1.
+            // The default literal conversion is an exception: (double)default would give the wrong value for M(double? x = default).
 
-            if (convertedExpression.ConstantValue == null && convertedExpression.Kind == BoundKind.Conversion)
+            if (convertedExpression.ConstantValue == null && convertedExpression.Kind == BoundKind.Conversion && convertedExpression.Syntax.Kind() != SyntaxKind.DefaultLiteralExpression)
```

Comment:
>convertedExpression.ConstantValue == null [](start = 16, length = 41)

I assume ```default(int?)``` doesn't pass this test due to a bug that the BoundDefaultExpression node has a constant value in that case.

Reply:
The default operator would not use "default literal" syntax. It uses default expression syntax. See ParseDefaultExpression.

Reply:
> The default operator would not use "default literal" syntax. It uses default expression syntax. See ParseDefaultExpression.

The condition ```convertedExpression.ConstantValue == null``` has nothing to do with syntax though.

Reply:
@AlekseyTs `default(int?)` does have `convertedExpression.ConstantValue == null`, but its `convertedExpression.Kind` is `BoundKind.DefaultExpression`, which is why the whole condition is `false`.